Web UI for AutoG (#27)
amazon-science/Automatic-Table-to-Graph-Generation: web_app/AWS_SETUP.md

> # AWS Credentials Setup for AutoG2 Web App

Reviewer: ```suggestion
# AWS Credentials Setup for Using AutoG Web App
```

## Overview

> The AutoG2 web app requires AWS credentials to access AWS Bedrock for LLM processing. For security reasons, credentials should be set up in your terminal environment before running the web app.

Reviewer: ```suggestion
The AutoG web app requires AWS credentials to access LLMs in AWS Bedrock. For security reasons, credentials should be set up in your terminal environment before running the web app.
```

## Setup Methods

### Method 1: Using AWS CLI (Recommended)

1. **Install AWS CLI** (if not already installed):
   ```bash
   # Ubuntu/Debian
   sudo apt-get install awscli
   
   # macOS
   brew install awscli
   
   # Or follow: https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html
   ```

2. **Configure AWS CLI**:
   ```bash
   aws configure
   ```
   
   You'll be prompted to enter:
   - AWS Access Key ID
   - AWS Secret Access Key
   - Default region name (e.g., `us-west-2`)
   - Default output format (e.g., `json`)

3. **Test your credentials**:
   ```bash
   aws sts get-caller-identity
   ```

### Method 2: Environment Variables

Set the following environment variables in your terminal:

```bash

> export AWS_ACCESS_KEY_ID=your_access_key_here

Reviewer: ```suggestion
export AWS_ACCESS_KEY_ID=<your_access_key_here>
```

> export AWS_SECRET_ACCESS_KEY=your_secret_key_here

Reviewer: ```suggestion
export AWS_SECRET_ACCESS_KEY=<your_secret_key_here>
```

> export AWS_SESSION_TOKEN=your_session_token_here  # Optional for temporary credentials

Reviewer: ```suggestion
export AWS_SESSION_TOKEN=<your_session_token_here>  # Optional for temporary credentials
```

> export AWS_DEFAULT_REGION=us-west-2

Reviewer: ```suggestion
export AWS_DEFAULT_REGION=<an_aws_region>
```

Author: This is the variable for default region.

```

### Method 3: Using the Setup Script

Run the provided setup script:

```bash
cd web_app
chmod +x setup_aws_credentials.sh
./setup_aws_credentials.sh
```

## Required Permissions

> Your AWS credentials need the following permissions for AutoG2:

Reviewer: ```suggestion
Your AWS credentials need the following permissions for AutoG:
```

- `bedrock:InvokeModel` - To call LLM models

> - Access to specific Bedrock models:

Reviewer: Suggest to only keep the Sonnet 4 and Sonnet 45 models, and remove the other model configuration.

And Add one paragraph to let users to know they can add additional models in the `webutils/config.py` file.

Author: I kept all these models same as in the original scripts. Should I remove them from the web app?

Reviewer: Yes. Let's remove unused model names.

  - `anthropic.claude-3-5-sonnet-20241022-v2:0`
  - `anthropic.claude-3-sonnet-20240229-v1:0`
  - `anthropic.claude-3-haiku-20240307-v1:0`

## Running the Web App

Once your AWS credentials are configured:

```bash
cd web_app

> python run_autog2_webapp.py

Reviewer: rename the file to `run_autog_webapp.py` to keep all namings consistent.
